src/libs/actions/Report.js

```diff
@@ -559,7 +522,29 @@ function subscribeToUserEvents() {
     // Live-update a report's actions when an 'edit comment' event is received.
     PusherUtils.subscribeToPrivateUserChannelEvent(Pusher.TYPE.REPORT_COMMENT_EDIT,
         currentUserAccountID,
-        pushJSON => updateReportActionMessage(pushJSON.reportID, pushJSON.sequenceNumber, pushJSON.message));
+        ({reportID, sequenceNumber, message}) => {
+            // We only want the active client to process these events once otherwise multiple tabs would decrement the 'unreadActionCount'
+            if (!ActiveClientManager.isClientTheLeader()) {
```

Comment:
This method is referring to multiple clients running on the same machine, correct? I went and read that file and it wasn't really clear to me why we have it.

Reply:
Things like Onyx subscriptions or Pusher subscriptions can exist in multiple tabs of a browser session (this is not an issue on native or tabs in different sessions). In some cases, updates to those channels should only be processed once and that's pretty much what this code does.

It might help to look at the few places where we are checking if the current client is the "leader". One example would be the processing of persisted requests when returning from offline. If two tabs both try to process the same data we would make 2x as many network requests as necessary when running queued requests. Another is browser notifications (without checking if the client is the leader we would end up with multiple notifications).

Reply:
Perfect explanation. Thank you!